nrrd/writer.py

```diff
@@ -252,30 +263,31 @@ def write(filename, data, header=None, detached_header=False, relative_data_path
 
         # If header & data in the same file is desired, write data in the file
         if not detached_header:
-            _write_data(data, fh, header, compression_level=compression_level)
+            _write_data(data, fh, header, compression_level=compression_level, index_order=index_order)
 
     # If detached header desired, write data to different file
     if detached_header:
         with open(data_filename, 'wb') as data_fh:
-            _write_data(data, data_fh, header, compression_level=compression_level)
+            _write_data(data, data_fh, header, compression_level=compression_level, index_order=index_order)
 
 
-def _write_data(data, fh, header, compression_level=None):
+def _write_data(data, fh, header, compression_level=None, index_order='A'):
+    if index_order not in ['F', 'C', 'A']:
+        raise NRRDError('Invalid index order')
+
     if header['encoding'] == 'raw':
         # Convert the data into a string
-        raw_data = data.tostring(order='F')
+        raw_data = data.tostring(order=index_order)
 
         # Write the raw data directly to the file
         fh.write(raw_data)
     elif header['encoding'].lower() in ['ascii', 'text', 'txt']:
-        # savetxt only works for 1D and 2D arrays, so reshape any > 2 dim arrays into one long 1D array
```

Comment:
Are there any reasons for using `savetxt` for 2D? I mostly removed it to simplify my solution but I can put it back.

Reply:
If I remember correctly, and it has been awhile, it's so it will make the ASCII data look like:

2D:
```
1 2 3
4 5 6
7 8 9
```

Would be nice to have to be consistent with NRRD spec, but not crucial. Maybe there's a cleaner way to set this up so that it works well with the index order.

Reply:
Makes sense. This should be an easy fix so I'll change it back.

Reply:
Makes sense so I will revert this. Making it work with the ordering should be an easy fix.